packages/wow-ui/src/hooks/useCheckedState.ts

```diff
@@ -13,17 +16,44 @@ interface CheckedStateProps {
 const useCheckedState = ({
   defaultChecked = false,
   checked: checkedProp,
-  disabled,
+  disabled: disabledProp,
+  value,
   onChange,
   onClick,
   onKeyDown,
 }: CheckedStateProps) => {
-  const [checked, setChecked] = useState<boolean>(() =>
-    checkedProp !== undefined ? checkedProp : defaultChecked
-  );
+  const group = useContext(MultiGroupContext);
+  const groupOnChange = group.onChange;
+  const groupChecked = group.checked?.includes(value);
+  const disabled = group.disabled || disabledProp || false;
 
+  const [checked, setChecked] = useState<boolean>(
+    groupChecked || checkedProp || defaultChecked
+  );
   const [pressed, setPressed] = useState<boolean>(false);
 
+  useEffect(() => {
+    if (groupChecked !== undefined) {
+      setChecked(groupChecked);
+    }
+  }, [groupChecked]);
+
+  useEffect(() => {
+    if (groupChecked === undefined && checkedProp !== undefined) {
+      setChecked(checkedProp);
+    }
+  }, [checkedProp, groupChecked]);
```

Comment:
p3;
중복되는 useEffect 내의 로직을 하나로 합쳐보면 어떨까요?!
```suggestion
  useEffect(() => {
    if (groupChecked !== undefined) {
      setChecked(groupChecked);
    } else if (checkedProp !== undefined) {
      setChecked(checkedProp);
    }
  }, [groupChecked, checkedProp]);
```

Reply:
두 부분을 합친다면 checkedProp이 변경되었을 때 불필요한 checked 상태값 변경이 일어나지 않을까 싶어서 저는 일단 분리를 해뒀어요!